File: Hackerrank/AppleandOrange.java
public class Solution {

    static void countApplesAndOranges(int s, int t, int a, int b, int[] apples, int[] oranges) {

        int appleCounter = 0;
        int orangeCounter = 0;

        for (int i = 0; i < apples.length; i++) {
            if (apples[i] + a >= s && apples[i] + a <= t)
                appleCounter++;
        }

        for (int i = 0; i < oranges.length; i++) {
            if (oranges[i] + b >= s && oranges[i] + b <= t)
                orangeCounter++;
        }

        System.out.println(appleCounter);
        System.out.println(orangeCounter);
    }
}

File: Hackerrank/apple-and-orange.py
#!/bin/python3
from functools import reduce

# Complete the countApplesAndOranges function below.
def countApplesAndOranges(s, t, a, b, apples, oranges):
    print(reduce(lambda count, x: count if a+x<s or a+x>t else count+1, apples, 0))
    print(reduce(lambda count, x: count if b+x<s or b+x>t else count+1, oranges, 0))

if __name__ == '__main__':
    st = input().split()
    s = int(st[0])
    t = int(st[1])
    ab = input().split()
    a = int(ab[0])
    b = int(ab[1])
    mn = input().split()
    m = int(mn[0])=
    n = int(mn[1])
    apples = list(map(int, input().rstrip().split()))
    oranges = list(map(int, input().rstrip().split()))
    countApplesAndOranges(s, t, a, b, apples, oranges)
